src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -145,44 +144,45 @@ public void SetDescriptionForRow(ColumnMappingDTO model)
          _setDescriptionForRow(model, values != null && values.All(v => v != model.ExcelColumn));
       }
 
-      public void UpdateMetaDataForModel()
+      public void UpdateMetaDataForModel(MetaDataFormatParameter mappingSource)
       {
-         if (_currentModel == null) return;
-
-         var metaData = _currentModel.Source as MetaDataFormatParameter;
-         if (metaData == null) return;
+         if (mappingSource == null)
+            return;
 
-         _mappings.First(m => m.MappingName == metaData.MetaDataId).ExcelColumn = _metaDataParameterEditorPresenter.Input;
-         metaData.ColumnName = _metaDataParameterEditorPresenter.Input;
-         metaData.IsColumn = false;
+         _mappings.First(m => m.MappingName == mappingSource.MetaDataId).ExcelColumn = _metaDataParameterEditorPresenter.Input;
+         mappingSource.ColumnName = _metaDataParameterEditorPresenter.Input;
+         mappingSource.IsColumn = false;
          ValidateMapping();
          _view.RefreshData();
          _view.CloseEditor();
       }
 
-      public void UpdateDescriptrionForModel()
+      public void UpdateDescriptionForModel(MappingDataFormatParameter mappingSource)
       {
-         if (!(_currentModel.Source is MappingDataFormatParameter))
-         {
+         if (mappingSource == null) 
             return;
-         }
 
-         var column = ((MappingDataFormatParameter)_currentModel.Source).MappedColumn;
+         var model = _mappings.FirstOrDefault(x => x.Source == mappingSource);
+
+         if (model == null) 
+            return;
+
+         var column = ((MappingDataFormatParameter)model.Source).MappedColumn;
          column.Unit = _mappingParameterEditorPresenter.Unit;
          if (!string.IsNullOrEmpty(column.Unit.ColumnName))
          {
             column.Unit = new UnitDescription(_rawData.GetColumn(column.Unit.ColumnName).First(u => !string.IsNullOrEmpty(u)), column.Unit.ColumnName);
```

Comment:
same question here with the First (I know it's not from this code change but since I see it :))

Reply:
same answer here: not found would mean we mapped a column that did not exist, I suggest we keeo the same behaviour as above.